src/rich-log/domain.hh

```diff
@@ -0,0 +1,131 @@
+#pragma once
+
+#include <clean-core/macros.hh>
+
+namespace rlog
+{
+namespace verbosity
+{
+enum type : int
+{
+    /// TRACE is used to provide diagnostic information verbosely and in very small steps
+    /// e.g. logging in each iteration of an algorithm
+    /// per default, trace should be compile-time DISABLED (as it can and will slow down the app)
+    Trace = 0,
+
+    /// DEBUG is used to provide diagnostic information that usually do not matter
+    /// e.g. logging the arguments to an algorithm
+    /// per default, debug should be compile-time ENABLED but runtime DISABLED
+    /// this means all message can be viewed post-mortem
```

Comment:
Does it? Right now it looks like runtime disabled messages are going nowhere

Reply:
lol excellent point. let's clarify tomorrow what we want. runtime disabled CAN be enabled runtime